### PR TITLE
Initial support for deno

### PR DESCRIPTION
**Pull request by redradist**
Initial support for modern Run-Time Deno https://deno.land/

### CODE REVIEW

**Comment by schmave** [comment]
This looks pretty cool! What are the differences between deno_bridge and node_bridge, other than that deno_bridge is written in TypeScript? Can deno run the existing node_bridge.js script? 

It seems unnecessary to have both a JS and TS version of the same script in the Brython repo, but perhaps I'm missing something. If there's a way to share more code between them I think that would be great. But maybe you could just explain a little bit more about your motivation for this change.

**Comment by redradist** [comment]
> 
> 
> This looks pretty cool! What are the differences between deno_bridge and node_bridge, other than that deno_bridge is written in TypeScript? Can deno run the existing node_bridge.js script?
> 
> It seems unnecessary to have both a JS and TS version of the same script in the Brython repo, but perhaps I'm missing something. If there's a way to share more code between them I think that would be great. But maybe you could just explain a little bit more about your motivation for this change.

@schmave `Deno` is other environment, it has different API in such case it is not possible to share implementation for this bridge (

File: scripts/deno_bridge.ts [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+// @ts-ignore
+__BRYTHON__.scope = __BRYTHON__.scope || {}
+// @ts-ignore
+__BRYTHON__.imported = __BRYTHON__.imported || {}
+// @ts-ignore
+__BRYTHON__.modules = __BRYTHON__.modules || {}
+// @ts-ignore
+__BRYTHON__.compile_python = $compile_python
+// @ts-ignore
+__BRYTHON__.import_hooks = $import_hooks
```
Comment (schmave) [inline scripts/deno_bridge.ts:182]: I see that $import_hooks was declared to be an async function, which means that it returns a Promise. I'm surprised that that would work with Brython, which expects __BRYTHON__.import_hooks to be a regular function. Are you sure that it works?

Comment (redradist) [reply scripts/deno_bridge.ts:182]: Whole this implementation is just first step to support Deno ... currently it is not working (

Comment (schmave) [reply scripts/deno_bridge.ts:182]: OK, I didn't realize this was still a work in progress. I'm inclined to close the pull request, then, until you've got it working. What do you think?

**Comment by schmave** [comment]
Closing for now. Feel free to open another once you've gotten it working.

**Comment by PierreQuentel** [comment]
Thanks Evan, I approve.

**Pull request closed without merging**